frontend/src/screens/apps/New.tsx

```diff
@@ -0,0 +1,231 @@
+
+  const reqMethodsParam = queryParams.get('request_methods');
+  const [requestMethods, setRequestMethods] = useState(reqMethodsParam ?? Object.keys(nip47MethodDescriptions).join(' '));
+
+  const maxAmountParam = queryParams.get('max_amount') ?? "";
+  const [maxAmount, setMaxAmount] = useState(parseInt(maxAmountParam));
+
+  const parseExpiresParam = (expiresParam: string): string => {
+    if (/\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}.\d{3}Z/.test(expiresParam)) {
+      const d = new Date(expiresParam); 
+      const isIso = d instanceof Date && !isNaN(d.getTime()) && d.toISOString() === expiresAtParam;
+      if (isIso) {
+        return expiresParam
+      }
+    }
+    if (!isNaN(parseInt(expiresParam))){
+      return (new Date(parseInt(expiresAtParam as string) * 1000)).toISOString()
+    }
+    return ""
+  }
+
+  // Only timestamp in seconds or ISO string is expected
+  const expiresAtParam = parseExpiresParam(queryParams.get('expires_at') ?? "")
+  const [expiresAt, setExpiresAt] = useState(expiresAtParam ?? "");
+  const [days, setDays] = useState(0);
+  const [expireOptions, setExpireOptions] = useState(false);
+
+  const today = new Date();
+  const handleDays = (days: number) => {
+    setDays(days);
+    if (!days) {
+      setExpiresAt("");
+      return;
+    }
+    const expiryDate = new Date(today.getTime() + days * 24 * 60 * 60 * 1000);
+    setExpiresAt(expiryDate.toISOString());
+  }
+
+  const handleRequestMethodChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const rm = event.target.value;
+    if (requestMethods.includes(rm)) {
+      // If checked and item is already in the list, remove it
+      const newMethods = requestMethods.split(" ").filter((reqMethod) => reqMethod !== rm).join(" ")
+      setRequestMethods(newMethods);
+    } else {
+      setRequestMethods(`${requestMethods} ${rm}`);
+    }
+  }
+
+  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    if (!info) return;
+    const formData = new FormData();
+    formData.append("name", appName);
+    formData.append("pubkey", pubkey);
+    formData.append("maxAmount", maxAmount.toString());
+    formData.append("budgetRenewal", budgetRenewal);
+    formData.append("expiresAt", expiresAt);
+    formData.append("requestMethods", requestMethods);
+    formData.append("returnTo", returnTo);
+    formData.append("_csrf", info.csrf);
+    try {
+      const response = await fetch("/api/apps", {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/x-www-form-urlencoded',
+        },
+        body: JSON.stringify(formData.toString()),
+      })
+      console.log(response)
+      // TODO: Navigate to success screen with data
+      // navigate("/apps/success");
+    } catch (error) {
+      // TODO: Deal with invalid pubkey format error
+      // Invalid expiresAt error
```

Comment:
can you address this TODO

Reply:
Maybe just a console log is fine for now actually? and just remove those comments